mailer/__init__.py

```
@@ -176,8 +226,81 @@ def worker():
+    else:
+        raise SystemExit("Excel missing 'email' column")
+
+    def first_name(rows) -> str | None:
+        for _, r in rows.iterrows():
+            v = r.get("vorname")
+            if pd.notna(v) and str(v).strip():
+                return str(v).strip()
+        return None
+
+    if "company" in leads.columns:
+        for company, group in leads.groupby("company"):
```

Ruff (0.11.9)

Loop control variable `company` not used within loop body

Rename unused `company` to `_company`

(B007)